SU2_CFD/include/drivers/CDriver.hpp

```diff
@@ -707,6 +723,112 @@ class CDriver {
    */
   map<string, string> GetAllBoundaryMarkersType();
 
+  /*!
+   * \brief Set the mesh displacement for the elasticity mesh solver.
+   * \param[in] iMarker - Marker identifier.
+   * \param[in] iVertex - Vertex identifier.
+   * \param[in] DispX - Value of the mesh displacement in the direction X.
+   * \param[in] DispY - Value of the mesh displacement in the direction Y.
+   * \param[in] DispZ - Value of the mesh displacement in the direction Z.
+   */
+  void SetMeshDisplacement(unsigned short iMarker, unsigned long iVertex, passivedouble DispX, passivedouble DispY, passivedouble DispZ);
+
+  /*!
+   * \brief Communicate the boundary mesh displacements in a python call
+   */
+  void CommunicateMeshDisplacement(void);
+
+  /*!
+   * \brief Return the sensitivities of the mesh boundary vertices.
+   * \param[in] iMarker - Marker identifier.
+   * \param[in] iVertex - Vertex identifier.
+   * \return Vector of sensitivities.
+   */
+  vector<passivedouble> GetMeshDisp_Sensitivity(unsigned short iMarker, unsigned short iVertex);
+
+  /*!
+   * \brief Set the load in X direction for the structural solver.
+   * \param[in] iMarker - Marker identifier.
+   * \param[in] iVertex - Vertex identifier.
+   * \param[in] LoadX - Value of the load in the direction X.
+   * \param[in] LoadX - Value of the load in the direction Y.
+   * \param[in] LoadX - Value of the load in the direction Z.
+   */
+  void SetFEA_Loads(unsigned short iMarker, unsigned short iVertex, passivedouble LoadX,
+                    passivedouble LoadY, passivedouble LoadZ);
+
+  /*!
+   * \brief Return the displacements from the FEA solver.
+   * \param[in] iMarker - Marker identifier.
+   * \param[in] iVertex - Vertex identifier.
+   * \return Vector of displacements.
+   */
+  vector<passivedouble> GetFEA_Displacements(unsigned short iMarker, unsigned short iVertex);
+
+  /*!
+   * \brief Return the velocities from the FEA Solver.
+   * \param[in] iMarker - Marker identifier.
+   * \param[in] iVertex - Vertex identifier.
+   * \return Vector of velocities.
+   */
+  vector<passivedouble> GetFEA_Velocity(unsigned short iMarker, unsigned short iVertex);
+
+  /*!
+   * \brief Return the velocities from the FEA Solver.
+   * \param[in] iMarker - Marker identifier.
+   * \param[in] iVertex - Vertex identifier.
+   * \return Vector of velocities at time n.
+   */
+  vector<passivedouble> GetFEA_Velocity_n(unsigned short iMarker, unsigned short iVertex);
```

Comment:
It would be good to have an architecture document about how this will work with python.

Reply:
Yep, I'll add some documentation soon, but I'd rather do it with a tutorial. I think I should be able to do it by the time we do the release.